src/Pages/Main/Main.js

```diff
@@ -0,0 +1,78 @@
+import { useState, useEffect } from "react";
+import SliderComponent from "../../Components/SliderComponent";
+import MainSlider from "./components/MainSlider";
+import InfiniteScroll from "../../Components/InfiniteScroll";
+import styled from "styled-components";
+import SubBanner from "./components/SubBanner";
+
+const Main = ({ theme }) => {
+  const [top10, setTop10] = useState([]);
+  const [planned, setPlanned] = useState([]);
+  const [updated, setUpdated] = useState([]);
+
+  useEffect(() => {
+    fetch("http://localhost:3000/Data/data.json")
+      .then((res) => res.json())
+      .then((res) => {
+        setTop10(res.top_10_data);
+        setPlanned(res.planned_data);
+        setUpdated(res.updated_data);
+      });
+  }, []);
+
+  const handleScroll = () => {
+    const scrollHeight = document.documentElement.scrollHeight;
+    const scrollTop = document.documentElement.scrollTop;
+    const clientHeight = document.documentElement.clientHeight;
+    if (scrollTop + clientHeight >= scrollHeight - 50) {
+      if (updated.length > 50) {
+        return;
+      }
+      setUpdated([...updated, ...updated]);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", () => handleScroll());
+    return () => {
+      window.removeEventListener("scroll", () => handleScroll());
+    };
+  });
```

Comment:
```suggestion
  useEffect(() => {
    window.addEventListener("scroll", handleScroll);
    return () => {
      window.removeEventListener("scroll", handleScroll);
    };
  }, [updated]);
```
- 모든 렌더 때마다 실행시킬 내용은 아닌 것 같습니다
- 화살표 함수로 써주지 않아도 됩니다